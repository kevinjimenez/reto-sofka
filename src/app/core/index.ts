export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
