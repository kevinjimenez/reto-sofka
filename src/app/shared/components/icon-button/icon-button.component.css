.container {
  background-color: transparent;
	border-radius: 0.2rem;
	border: none;
	display: flex;
	align-items: center;
	justify-content: center;
  cursor: pointer;
}

