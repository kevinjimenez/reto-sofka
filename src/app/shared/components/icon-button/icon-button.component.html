<button class="container" [type]="buttonType()" (click)="onClick.emit($event)">
	<img [src]="icon() | image: 'icons'" [alt]="icon()" />
</button>
