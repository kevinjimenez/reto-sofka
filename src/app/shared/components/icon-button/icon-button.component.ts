import { Component, input, output } from '@angular/core';
import { ImagePipe } from '../../pipes/image.pipe';

@Component({
	selector: 'app-icon-button',
	standalone: true,
	imports: [ImagePipe],
	templateUrl: './icon-button.component.html',
	styleUrl: './icon-button.component.css'
})
export class IconButtonComponent {
	public icon = input.required<string>();

	public buttonType = input<'button' | 'submit'>('button');
	// @Output() public onClick: EventEmitter<void> = new EventEmitter();
	public onClick = output<MouseEvent>();
}
