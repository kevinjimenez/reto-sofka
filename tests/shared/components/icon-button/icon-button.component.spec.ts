import { ComponentRef } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ImagePipe } from '../../../../src/app/shared/pipes/image.pipe';
import { IconButtonComponent } from '../../../../src/app/shared/components/icon-button/icon-button.component';

describe('IconButtonComponent', () => {
	let component: IconButtonComponent;
	let componentRef: ComponentRef<IconButtonComponent>;
	let fixture: ComponentFixture<IconButtonComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [IconButtonComponent, ImagePipe]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(IconButtonComponent);
		component = fixture.componentInstance;
		componentRef = fixture.componentRef;
		componentRef.setInput('icon', 'test');

		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should render icon button', () => {
		componentRef.setInput('icon', 'test');
		fixture.detectChanges();

		const button = compiled.querySelector('button');
		const img = compiled.querySelector('img') as HTMLImageElement;
		expect(button).toBeTruthy();
		expect(img).toBeTruthy();
		expect(img.alt).toBe('test');
	});
});
