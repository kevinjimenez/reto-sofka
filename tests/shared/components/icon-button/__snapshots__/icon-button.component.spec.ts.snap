// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <button
    class="container"
    type="button"
  >
    <img
      alt="test"
      src="assets/icons/test"
    />
  </button>
</div>
`;
